tests/test_product_mixin.py

```diff
@@ -85,3 +79,43 @@ def test_it_visualized_multispectral_to_rgb_with_no_product(raster):
     result = raster.visualize('TrueColor')
     assert isinstance(result, GeoRaster2)
     assert result.band_names == ['red', 'green', 'blue']
+
+
+def test_custom_product_preserve_mask():
+
+    def foo(red, green):
+        arr1 = red + 1
+        arr2 = green + 2
+        arr = np.stack([arr1[0], arr2[0]])
+        return arr
+
+    raster = multi_raster_with_no_data()
+    output = raster.apply(sensor_bands_info(), 'custom', function=foo, required_bands=['red', 'green'],
```

Comment:
stating it is a custom  product can be inferred by the existence of the function method